### PR TITLE
[Deprecated API] remove `version`

### PR DESCRIPTION
**Pull request by zhaixuejun1993**
### Details:
- Deprecated ov::DiscreteTypeInfo member version
- Remove related reference or using about version

### Tickets:
 - 106458 (100091)

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
@ilyachur will string compare instead of integers affect performance of transformations / etc ?

**Comment by ilyachur** [comment]
> @ilyachur will string compare instead of integers affect performance of transformations / etc ?

Transformations shouldn't compare strings... Instead of this, they can compare `DiscreteTypeInfo` inside these structures we use hash for comparison in order to minimize the performance impact.
Any way if at the current moment some transformations use string comparison, I propose to change it separately.

File: src/core/include/openvino/core/node.hpp [hunk 1]
```diff
@@ -409,12 +409,8 @@ class OPENVINO_API Node : public std::enable_shared_from_this<Node> {
     /// Get all the nodes that uses the current node
     NodeVector get_users(bool check_is_used = false) const;
 
-    /// \return Version of this node
-    OPENVINO_DEPRECATED("This method is deprecated and will be removed soon.")
-    virtual size_t get_version() const {
-        OPENVINO_SUPPRESS_DEPRECATED_START
-        return get_type_info().version;
-        OPENVINO_SUPPRESS_DEPRECATED_END
+    virtual std::shared_ptr<Node> get_default_value() const {
```
Comment (ilyachur) [inline src/core/include/openvino/core/node.hpp:412]: Why did you add the new (old removed) method?

File: src/core/tests/type_info.cpp [hunk 2]
```diff
@@ -46,40 +46,37 @@ TEST(type_info, compare_new_type) {
 }
 
 TEST(type_info, compare_new_with_old_type) {
-    ov::DiscreteTypeInfo type1("type1", static_cast<uint64_t>(0), "version1");
-    ov::DiscreteTypeInfo type1_o("type1", static_cast<uint64_t>(0));
+    ov::DiscreteTypeInfo type1("type1", "version1");
+    ov::DiscreteTypeInfo type1_o("type1", "version1");
```
Comment (ilyachur) [inline src/core/tests/type_info.cpp:50]: I think we can simple remove this test

File: src/core/tests/type_info.cpp [hunk 3]
```diff
@@ -10,11 +10,11 @@
...
 OPENVINO_SUPPRESS_DEPRECATED_START
 TEST(type_info, compare_old_type) {
-    ov::DiscreteTypeInfo type1("type1", static_cast<uint64_t>(0));
-    ov::DiscreteTypeInfo type2("type2", static_cast<uint64_t>(0));
-    ov::DiscreteTypeInfo type3("type1", 1ul);
-    ov::DiscreteTypeInfo type4("type3", static_cast<uint64_t>(0), &type1);
-    ov::DiscreteTypeInfo type5("type3", static_cast<uint64_t>(0), &type2);
+    ov::DiscreteTypeInfo type1("type1");
+    ov::DiscreteTypeInfo type2("type2");
+    ov::DiscreteTypeInfo type3("type1");
```
Comment (ilyachur) [inline src/core/tests/type_info.cpp:15]: Please change the test and remove redundant types

File: src/core/tests/type_info.cpp [hunk 4]
```diff
@@ -46,40 +46,37 @@ TEST(type_info, compare_new_type) {
...
-    EXPECT_EQ(hash_val(type.name, type.version_id, type.version), type.hash());
-    EXPECT_EQ(hash_val(type_old.name, type_old.version_id, type_old.version), type_old.hash());
-    EXPECT_EQ(hash_val(type_with_version.name, type_with_version.version_id, type_with_version.version),
-              type_with_version.hash());
-    EXPECT_EQ(hash_val(type_empty_name.name, type_empty_name.version_id, type_empty_name.version),
-              type_empty_name.hash());
-    EXPECT_EQ(hash_val(type_empty_ver.name, type_empty_ver.version_id, type_empty_ver.version), type_empty_ver.hash());
+    ov::DiscreteTypeInfo type("type1", "version1");
+    ov::DiscreteTypeInfo type_old("type1");
+    ov::DiscreteTypeInfo type_with_version("type1", "version1");
```
Comment (ilyachur) [inline src/core/tests/type_info.cpp:63]: I think we need to remove redundant types

**Comment by ilyachur** [comment]
@zhaixuejun1993 Let's wait for CI. Test changes can be done in next PR

File: src/core/tests/opset.cpp [hunk 5]
```diff
@@ -121,7 +121,7 @@ TEST(opset, custom_opset) {
     opset.insert<MyOpNewFromOld>();
     opset.insert<MyOpNew>();
     EXPECT_EQ(opset.get_types_info().size(), 3);
-    EXPECT_TRUE(opset.contains_type("MyOpNewFromOld"));
-    EXPECT_TRUE(opset.contains_type("MyOpNew"));
-    EXPECT_TRUE(opset.contains_type("MyOpIncorrect"));
+    EXPECT_TRUE(opset.contains_type(std::string("MyOpNewFromOld")));
```
Comment (ilyachur) [inline src/core/tests/opset.cpp:124]: Why do we need this change?

**Review by ilyachur — APPROVED** [review]
